internal/controllers/topology/cluster/patches/variables/variables.go

```diff
@@ -180,11 +180,13 @@ func Global(clusterTopology *clusterv1.Topology, cluster *clusterv1.Cluster) ([]
 		if variable.Name == BuiltinsName {
 			continue
 		}
-
-		variables = append(variables, runtimehooksv1.Variable{
-			Name:  variable.Name,
-			Value: variable.Value,
-		})
+		// Add the variable only if it is defined for the current patch, or it is defined for all patches.
+		if variable.DefinitionFrom == patchName || variable.DefinitionFrom == allPatchesDefinitionFrom {
```

Comment:
`variable.DefinitionFrom == allPatchesDefinitionFrom`

I think this is not good enough. This way an extension also gets variable values that have been defined by other external patches.

The idea was to have a forcing function that every runtime extension must define all the variables it uses (except builtins) and then it only gets those

Reply:
I think we have to do roughly the following:
* iterate through ClusterClass.status.variables to get all variables defined by a patch
  * => this gives us a set of variable names that this patch should receive
* iterate through variable values
  * send the variable value if:
    * the variable value name is in the set of names and
    * there's either a global variable with that name or one specific for this "definitionFrom" (as there cannot be both this should be good enough, and I think that is your current check in l.184)

Reply:
I think the latest version is following this spec.

Reply:
Looks good!